textpipe/doc.py

```diff
@@ -431,15 +434,64 @@ def vectors(self):
         vector: The vector representation of the word
 
         >>> doc = Doc('Test sentence')
-        >>> doc.vectors['Test'][2]
+        >>> doc.word_vectors['Test']['is_oov']
         True
-        >>> doc.vectors['Test'][3].shape[-1]
+        >>> doc.word_vectors['Test']['vector'].shape[-1]
         384
-        >>> doc.vectors['Test'][1] == doc.vectors['sentence'][1]
+        >>> doc.word_vectors['Test']['vector_norm'] == doc.word_vectors['sentence']['vector_norm']
         False
         """
-        return {token.text: (token.has_vector,
-                             token.vector_norm,
-                             token.is_oov,
-                             token.vector)
+        return {token.text: {'has_vector': token.has_vector,
+                             'vector_norm': token.vector_norm,
+                             'is_oov': token.is_oov,
+                             'vector': token.vector}
```

Comment:
What's the type of the vector, and does this type serialize and deserialize to and from JSON?

Reply:
The vectors are ndarrays which are not directly serializable to JSON with a JSON dump directly. I'll change the code such that they will be returned as lists (by using `.tolist()`). In this way they can be serialized to JSON and deserialised by changing them back to numpy arrays (e.g. for the doc vector: `numpy.array(json.dumps(doc.vector.tolist()))`.